pkg/filesystem/path/parser.go

```diff
@@ -0,0 +1,16 @@
+package path
+
+// Parser is used by Resolve to parse paths in the resolution.
+// The Parser implementations will be used by value in several bookkeeping
+// structures, they must be implemented so they can safely be given as values.
```

Comment:
I think "safely be given as values" is not a very clear way of phrasing it. I would recommend that we state something like this instead:

> Implementations of ParseScope() should return a new copy of Parser and leave the current instance unmodified. It is permitted to call ParseScope() multiple times.

The same for RelativeParser.

Reply:
Sounds a lot better!